Ginger/GingerCoreNET/Run/RunListenerLib/SealightsExecutionLogger/SealightsReportExecutionLogger.cs

```diff
@@ -0,0 +1,207 @@
+                return mSealightsApiHandler;
+            }
+        }
+
+        public SealightsReportExecutionLogger(Context context)
+        {
+            mContext = context;
+        }
+
+        #region RunSet
+        public void RunSetStart(RunSetConfig runsetConfig)
+        {
+            try
+            {
+                Reporter.ToStatus(eStatusMsgKey.PublishingToCentralDB, "Sealights Session Creation");
+                if (WorkSpace.Instance.RunsetExecutor != null && WorkSpace.Instance.RunsetExecutor.RunSetConfig != null
+                    && WorkSpace.Instance.RunsetExecutor.RunSetConfig.ExecutionID != null)
+                {
+                    runsetConfig.ExecutionID = WorkSpace.Instance.RunsetExecutor.RunSetConfig.ExecutionID;
+                }
+                SealightsReportApiHandler.SendCreationTestSessionToSealightsAsync();
+            }
+            catch(Exception err)
+            {
+                string error = err.Message;
+            }
+        }
+
+    
+
+        public async Task RunSetEnd(RunSetConfig runsetConfig)
+        {
+            await SealightsReportApiHandler.SendDeleteSessionToSealightsAsync(); // Delete Sealights session
+        }
+
+        #endregion RunSet   
+
+
+        #region Runner
+        public override async void RunnerRunStart(uint eventTime, GingerRunner gingerRunner, bool offlineMode = false)
+        {
+            //await RunnerRunStartTask(gingerRunner);
+        }
+
+        public override async void RunnerRunEnd(uint eventTime, GingerRunner gingerRunner, string filename = null, int runnerCount = 0, bool offlineMode = false)
+        {
+            if (!gingerRunner.Active || gingerRunner.Status == Execution.eRunStatus.Blocked)
+            {
+                return;
+            }            
+        }
+
+        #endregion Runner
+
+        #region BusinessFlow
+        public override async void BusinessFlowStart(uint eventTime, BusinessFlow businessFlow, bool ContinueRun = false)
+        {
+           //await BusinessFlowStartTask(businessFlow);
+        }
+
+        
+
+        public override async void BusinessFlowEnd(uint eventTime, BusinessFlow businessFlow, bool offlineMode = false)
+        {
+            if (SealightsReportApiHandler.TestSessionId == null) // this will make sure not reporting to Sealight for partial execution done from Run set page
+            {
+                return;
+            }
+
+            if (WorkSpace.Instance.Solution.LoggerConfigurations.SealightsReportedEntityLevel == eSealightsEntityLevel.BusinessFlow)
+            {
+                if (!businessFlow.Active || businessFlow.RunStatus == Execution.eRunStatus.Blocked)
+                {
+                    return;
+                }
+
+                await SealightsReportApiHandler.SendingTestEventsToSealightsAsync(businessFlow.Name, businessFlow.StartTimeStamp, businessFlow.EndTimeStamp, businessFlow.Status.ToString());
+            }
```

Comment:
Create mapping table for all statuses to make sure we send only PAss/Fail/Skipped